index.js

```diff
@@ -46,13 +54,27 @@ function Streampub (opts) {
 }
 inherits(Streampub, Transform)
 
+// Theoretically this could be done in _flush, but somehow this generates an error
+// this is a workaround for that error ('no write after end')
+Streampub.prototype._end = Streampub.prototype.end;
```

Comment:
I discuss this a bit more below, but this should definitely be done in flush. If `_generateCover` uses `self._transform` instead of `self.write` then it should work fine. No need to wait on the callback even.